pkg/upload/xcode-archive.go

```diff
@@ -0,0 +1,102 @@
+package upload
+
+import (
+	"fmt"
+	"github.com/bugsnag/bugsnag-cli/pkg/ios"
+	"github.com/bugsnag/bugsnag-cli/pkg/log"
+	"github.com/bugsnag/bugsnag-cli/pkg/options"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
+	"os"
+	"path/filepath"
+)
+
+// ProcessXcodeArchive processes an Xcode archive, locating its dSYM files and uploading them
+// to a Bugsnag server.
+//
+// Parameters:
+// - options: CLI options provided by the user, including Xcode archive settings.
+// - endpoint: The server endpoint for uploading dSYM files.
+// - logger: Logger instance for logging messages during processing.
+//
+// Returns:
+// - An error if any part of the process fails, otherwise nil.
+func ProcessXcodeArchive(options options.CLI, endpoint string, logger log.Logger) error {
+	xcarchiveOptions := options.Upload.XcodeArchive
+	var (
+		xcarchivePath string
+		plistPath     string
+		dwarfInfo     []*ios.DwarfInfo
+		tempDirs      []string
+		tempDir       string
+		err           error
+	)
+
+	// Ensure temporary directories are cleaned up after execution
+	defer func() {
+		for _, tempDir := range tempDirs {
+			_ = os.RemoveAll(tempDir)
+		}
+	}()
+
+	// Search for an .xcarchive in the specified paths
+	for _, path := range xcarchiveOptions.Path {
+		if filepath.Ext(path) == ".xcarchive" {
+			xcarchivePath = path
+		} else if utils.IsDir(path) {
+			logger.Info(fmt.Sprintf("Searching for Xcode Archives in %s", path))
+			if ios.IsPathAnXcodeProjectOrWorkspace(path) {
+				xcarchiveOptions.ProjectRoot = ios.GetDefaultProjectRoot(path, xcarchiveOptions.ProjectRoot)
+				logger.Info(fmt.Sprintf("Setting `--project-root` from Xcode project settings: %s", xcarchiveOptions.ProjectRoot))
+
+				if xcarchiveOptions.Scheme == "" {
+					xcarchiveOptions.Scheme, err = ios.GetDefaultScheme(path)
+					if err != nil {
+						return fmt.Errorf("Error determining default scheme: %w", err)
+					}
+				}
+
+				archiveLocation, err := ios.GetXcodeArchiveLocation()
+				if err != nil {
+					logger.Warn(fmt.Sprintf("Error getting Xcode archive location: %s", err))
+					return err
+				}
+				xcarchivePath, err = ios.GetLatestXcodeArchive(archiveLocation, xcarchiveOptions.Scheme)
+				if err != nil {
+					return fmt.Errorf("Error locating latest .xcarchive: %w", err)
+				}
+			} else {
+				return fmt.Errorf("No .xcarchive found in %s", path)
+			}
+		}
```

Comment:
Would probably be suitable to have a log info here to confirm the archive file we've picked

Reply:
I've adjusted the looping through the provided paths to match other functions which now means the log for showing what file we're using is included.